app/models/exercise/approach.rb

```diff
@@ -31,4 +41,28 @@ def git = Git::Exercise::Approach.new(slug, exercise.slug, exercise.git_type, sy
 
   memoize
   def content_html = Markdown::Parse.(content)
+
+  def matching_tags?(check_tags)
+    return false if check_tags.empty?
+
+    all_tags = tag_conditions[:all].to_a
+    return false if all_tags.present? && (all_tags - check_tags).present?
+
+    any_tags = tag_conditions[:any].to_a
+    return false if any_tags.present? && (any_tags & check_tags).empty?
```

Comment:
So in the case where someone specifies `all` and `any`, both are honoured? That's probably right, but I just want to check :)

Reply:
Yes, that's according to the spec I think